pkg/platform/src/runtime/node.ts

```diff
@@ -160,12 +160,17 @@ export async function build(
           ),
         }),
       );
-      const cmd = ["npm", "install", "--force"];
+      const cmd = [
+        "npm install",
+        "--force",
+        "--platform=linux",
+        input.architecture === "arm64" ? "--arch=arm64" : "--arch=x64",
+        // support npm versions 10 and above
+        "--os=linux",
+        input.architecture === "arm64" ? "--cpu=arm64" : "--cpu=x64",
+      ];
       if (installPackages.includes("sharp")) {
-        cmd.push(
-          "--os=linux --libc=glibc",
-          input.architecture === "arm64" ? "--arch=arm64" : "--arch=x64",
-        );
+        cmd.push("--libc=glibc");
```

Comment:
I'm not sure if it's really worth using the `-libc` flag just for `sharp`. I expect it to work even without that line.

https://github.com/lovell/sharp/issues/2230#issuecomment-1509870904

Reply:
![image](https://github.com/sst/ion/assets/41789633/0f4b5a2b-4c9e-48ee-8ce5-22ef1750030a)
It will result in exactly the same binaries whether the flag is used or not.

```sh
$ npm install --cpu=x64 --os=linux sharp
$ npm install --cpu=x64 --os=linux --libc=glibc sharp
```